src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationAndGeneratorDriverTranslationAction_Actions.cs

```diff
@@ -35,6 +35,17 @@ public override Task<Compilation> TransformCompilationAsync(Compilation oldCompi
                 // Replacing a single tree doesn't impact the generated trees in a compilation, so we can use this against
                 // compilations that have generated trees.
                 public override bool CanUpdateCompilationWithStaleGeneratedTreesIfGeneratorsGiveSameOutput => true;
+
+                public override CompilationAndGeneratorDriverTranslationAction? TryMergeWithPrior(CompilationAndGeneratorDriverTranslationAction priorAction)
+                {
+                    if (priorAction is TouchDocumentAction priorTouchAction &&
+                        priorTouchAction._newState == this._oldState)
```

Comment:
💭 Is it possible to end up in a state where two `TouchDocumentAction` instances are separated by something else? In such a case, is it sufficient to simply ignore all but the last `TouchDocumentAction`?

Reply:
@sharwell: good question! Yep, this could be made smarter. A case where you have a Solution snapshot, and keep forking it with alternate edits to A.cs and B.cs won't consolidate those together with this, even though the earlier replacements don't really matter. There's some icky edge cases though more generally with how the code works. Consider a case where the user does the following changes, forking a Solution each time.

1. Replaces the contents of A.cs.
2. Removes A.cs.
3. Adds A.cs back with the same document ID.
4. Replaces the contents of A.cs again.

(Is this sense of operations sensical? No. Which probably means there's code out there that does it. 😄 )

Clearly the change in step 1 doesn't matter, however when we run the translation action for the second step, it's implemented as "remove the syntax tree that should be there for A.cs", but if step 1 doesn't run, it'd be the wrong tree. A smarter algorithm would probably be "we can merge any two TouchDocumentActions as long as we don't move them across any other action", but I'd want to see a case where that mattered before we wrote the fancier code and tests.